### PR TITLE
[Exporter] Added support for Database Instance resource (aka Lakebase)

### PR DESCRIPTION
**Pull request by alexott**
## Changes

Also adding more generic functions, i.e. handle of effective fields, etc.

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [x] using Go SDK
- [x] using TF Plugin Framework
- [x] has entry in `NEXT_CHANGELOG.md` file

### CODE REVIEW

File: exporter/abstractions_test.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package exporter
```
Comment (rauchy) [inline exporter/abstractions_test.go:1]: Is this placeholder file intentional? If there are no tests yet for abstractions.go, might be worth either adding tests for `copyEffectiveFieldsToInputFieldsWithConverters` or removing the empty file.

**Review by rauchy — APPROVED** [review]

File: exporter/abstractions_test.go (hunk 1 above)
Comment (alexott) [reply exporter/abstractions_test.go:1]: Ok, I'll check with the next PRs - I think that it's used there. If not, will remove it
